commit c428fec5a19817eb2332843686399c3c2f292a3e
Author: Alexander Kurtakov <akurtako@redhat.com>
Date:   Wed Feb 19 10:48:21 2014 +0200

    Bug 428512 - Adapt Java 1.5 in examples
    
    * Set BREE in .classpath where not set
    * Use foreach
    * Add missing Override
    
    Change-Id: I70338b87613a2eb960224aecbbcf9920b52c0d4c
    Signed-off-by: Alexander Kurtakov <akurtako@redhat.com>

3	3	examples/org.eclipse.swt.examples.browser/.classpath
6	7	examples/org.eclipse.swt.examples.browser/.settings/org.eclipse.jdt.core.prefs
3	0	examples/org.eclipse.swt.examples.browser/src/org/eclipse/swt/examples/browserexample/BrowserView.java
4	8	examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/LauncherPlugin.java
3	8	examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/SplitLayout.java
3	3	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CTabFolderTab.java
5	6	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
1	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
32	33	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
1	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
8	9	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
15	16	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
2	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
1	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
3	4	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CustomFontTab.java
10	17	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/GraphicsExample.java
2	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
2	2	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaLineStyler.java
3	3	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/layoutexample/LayoutExample.java
diff --git a/examples/org.eclipse.swt.examples.browser/.classpath b/examples/org.eclipse.swt.examples.browser/.classpath
index 7b14e15..64c5e31 100644
--- a/examples/org.eclipse.swt.examples.browser/.classpath
+++ b/examples/org.eclipse.swt.examples.browser/.classpath
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.5"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="src" path="src"/>
 	<classpathentry kind="output" path="bin"/>
-</classpath>
\ No newline at end of file
+</classpath>
diff --git a/examples/org.eclipse.swt.examples.browser/.settings/org.eclipse.jdt.core.prefs b/examples/org.eclipse.swt.examples.browser/.settings/org.eclipse.jdt.core.prefs
index dc37a46..416f4fb 100644
--- a/examples/org.eclipse.swt.examples.browser/.settings/org.eclipse.jdt.core.prefs
+++ b/examples/org.eclipse.swt.examples.browser/.settings/org.eclipse.jdt.core.prefs
@@ -1,12 +1,11 @@
-#Mon Dec 11 15:14:40 EST 2006
 eclipse.preferences.version=1
-org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=disabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.2
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.5
 org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
-org.eclipse.jdt.core.compiler.compliance=1.4
+org.eclipse.jdt.core.compiler.compliance=1.5
 org.eclipse.jdt.core.compiler.debug.lineNumber=generate
 org.eclipse.jdt.core.compiler.debug.localVariable=generate
 org.eclipse.jdt.core.compiler.debug.sourceFile=generate
-org.eclipse.jdt.core.compiler.problem.assertIdentifier=warning
-org.eclipse.jdt.core.compiler.problem.enumIdentifier=warning
-org.eclipse.jdt.core.compiler.source=1.3
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.5
diff --git a/examples/org.eclipse.swt.examples.browser/src/org/eclipse/swt/examples/browserexample/BrowserView.java b/examples/org.eclipse.swt.examples.browser/src/org/eclipse/swt/examples/browserexample/BrowserView.java
index 54f6ce0..cb034e6 100644
--- a/examples/org.eclipse.swt.examples.browser/src/org/eclipse/swt/examples/browserexample/BrowserView.java
+++ b/examples/org.eclipse.swt.examples.browser/src/org/eclipse/swt/examples/browserexample/BrowserView.java
@@ -29,6 +29,7 @@ public class BrowserView extends ViewPart {
 	 * 
 	 * @see ViewPart#createPartControl
 	 */
+	@Override
 	public void createPartControl(Composite frame) {
 		instance = new BrowserExample(frame, true);
 	}
@@ -38,6 +39,7 @@ public class BrowserView extends ViewPart {
 	 * 
 	 * @see org.eclipse.ui.part.ViewPart#setFocus
 	 */
+	@Override
 	public void setFocus() {
 		instance.focus();
 	}
@@ -45,6 +47,7 @@ public class BrowserView extends ViewPart {
 	/**
 	 * Called when the View is to be disposed
 	 */	
+	@Override
 	public void dispose() {
 		instance.dispose();
 		instance = null;
diff --git a/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/LauncherPlugin.java b/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/LauncherPlugin.java
index 4be6030..6a54470 100644
--- a/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/LauncherPlugin.java
+++ b/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/LauncherPlugin.java
@@ -110,8 +110,7 @@ public class LauncherPlugin extends AbstractUIPlugin {
 	 */
 	public static void freeResources() {
 		if (images != null) {
-			for (int i = 0; i < images.length; ++i) {
-				final Image image = images[i];
+			for (Image image : images) {
 				if (image != null) image.dispose();
 			}
 			images = null;
@@ -181,8 +180,7 @@ public class LauncherPlugin extends AbstractUIPlugin {
 		
 		/* Collect all launch categories -- coalesce those with same ID */
 		HashMap<String, ItemTreeNode> idMap = new HashMap<String, ItemTreeNode>();
-		for (int i = 0; i < configurationElements.length; ++i) {
-			final IConfigurationElement ce = configurationElements[i];
+		for (IConfigurationElement ce: configurationElements) {
 			final String ceName = ce.getName();
 			final String attribId = getItemAttribute(ce, LAUNCH_ITEMS_XML_ATTRIB_ID, null);
 			
@@ -197,8 +195,7 @@ public class LauncherPlugin extends AbstractUIPlugin {
 		
 		/* Generate launch category hierarchy */
 		Set<String> tempIdSet = new HashSet<String>(); // used to prevent duplicates from being entered into the tree
-		for (int i = 0; i < configurationElements.length; ++i) {
-			final IConfigurationElement ce = configurationElements[i];
+		for (IConfigurationElement ce : configurationElements) {
 			final String ceName = ce.getName();
 			final String attribId = getItemAttribute(ce, LAUNCH_ITEMS_XML_ATTRIB_ID, null);
 			
@@ -211,8 +208,7 @@ public class LauncherPlugin extends AbstractUIPlugin {
 		}
 		
 		/* Generate program tree */
-		for (int i = 0; i < configurationElements.length; ++i) {
-			final IConfigurationElement ce = configurationElements[i];
+		for (IConfigurationElement ce : configurationElements) {
 			final String ceName = ce.getName();
 			final String attribId = getItemAttribute(ce, LAUNCH_ITEMS_XML_ATTRIB_ID, null);
 
diff --git a/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/SplitLayout.java b/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/SplitLayout.java
index a2b522f..dad174d 100644
--- a/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/SplitLayout.java
+++ b/examples/org.eclipse.swt.examples.launcher/src/org/eclipse/swt/examples/launcher/SplitLayout.java
@@ -95,8 +95,7 @@ public class SplitLayout extends Layout {
 		clientArea.height -= marginBottom + marginTop;
 		Point position = new Point(clientArea.x, clientArea.y);
 
-		for (int i = 0; i < children.length; ++i) {
-			final Control child = children[i];
+		for (Control child : children) {
 			final Rectangle bounds;
 			if (splitDirection == splitHorizontally) {
 				int height = clientArea.height / children.length;
@@ -116,9 +115,7 @@ public class SplitLayout extends Layout {
 	private Point computeHSplitSize(Composite composite, int wHint, int hHint, boolean flushCache) {
 		Point size = new Point(marginLeft + marginRight, marginTop + marginBottom);
 		Control[] children = composite.getChildren();
-		for (int i = 0; i < children.length; ++i) {
-			final Control child = children[i];
-
+		for (Control child : children) {
 			Point childSize = child.computeSize(wHint, hHint, flushCache);
 			size.x = Math.max(size.x, childSize.x);
 			size.y += childSize.y + spacing;
@@ -129,9 +126,7 @@ public class SplitLayout extends Layout {
 	private Point computeVSplitSize(Composite composite, int wHint, int hHint, boolean flushCache) {
 		Point size = new Point(marginLeft + marginRight, marginTop + marginBottom);
 		Control[] children = composite.getChildren();
-		for (int i = 0; i < children.length; ++i) {
-			final Control child = children[i];
-
+		for (Control child : children) {
 			Point childSize = child.computeSize(wHint, hHint, flushCache);
 			size.x += childSize.x + spacing;
 			size.y = Math.max(size.y, childSize.y);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CTabFolderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CTabFolderTab.java
index 77cf9ee..6e9cde6 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CTabFolderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CTabFolderTab.java
@@ -417,11 +417,11 @@ class CTabFolderTab extends Tab {
 	void setImages () {
 		boolean setImage = imageButton.getSelection ();
 		CTabItem items[] = tabFolder1.getItems ();
-		for (int i = 0; i < items.length; i++) {
+		for (CTabItem item : items) {
 			if (setImage) {
-				items[i].setImage (instance.images[ControlExample.ciClosedFolder]);
+				item.setImage (instance.images[ControlExample.ciClosedFolder]);
 			} else {
-				items[i].setImage (null);
+				item.setImage (null);
 			}
 		}
 		setExampleWidgetSize ();
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
index 47d6d63..bfaa163 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
@@ -54,11 +54,11 @@ public class ControlExample {
 		initResources();
 		tabFolder = new TabFolder (parent, SWT.NONE);
 		tabs = createTabs();
-		for (int i=0; i<tabs.length; i++) {
+		for (Tab tab : tabs) {
 			TabItem item = new TabItem (tabFolder, SWT.NONE);
-		    item.setText (tabs [i].getTabText ());
-		    item.setControl (tabs [i].createTabFolderPage (tabFolder));
-		    item.setData (tabs [i]);
+		    item.setText (tab.getTabText ());
+		    item.setControl (tab.createTabFolderPage (tabFolder));
+		    item.setData (tab);
 		}
 		
 		/* Workaround: if the tab folder is wider than the screen,
@@ -133,8 +133,7 @@ public class ControlExample {
 	 */
 	void freeResources() {
 		if (images != null) {
-			for (int i = 0; i < images.length; ++i) {
-				final Image image = images[i];
+			for (Image image : images) {
 				if (image != null) image.dispose();
 			}
 			images = null;
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
index bf9a827..c4245a4 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
@@ -185,8 +185,7 @@ class CoolBarTab extends Tab {
 
 		/* Set the sizes after adding all cool items */
 		CoolItem[] coolItems = coolBar.getItems();
-		for (int i = 0; i < coolItems.length; i++) {
-			CoolItem coolItem = coolItems[i];
+		for (CoolItem coolItem : coolItems) {
 			Control control = coolItem.getControl();
 			Point size = control.computeSize(SWT.DEFAULT, SWT.DEFAULT);
 			Point coolSize = coolItem.computeSize(size.x, size.y);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
index cce1edd..d9b5769 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
@@ -271,18 +271,18 @@ abstract class Tab {
 			}
 		};
 		Control [] children = styleGroup.getChildren ();
-		for (int i=0; i<children.length; i++) {
-			if (children [i] instanceof Button) {
-				Button button = (Button) children [i];
+		for (Control child : children) {
+			if (child instanceof Button) {
+				Button button = (Button) child;
 				button.addSelectionListener (selectionListener);
 			} else {
-				if (children [i] instanceof Composite) {
+				if (child instanceof Composite) {
 					/* Look down one more level of children in the style group. */
-					Composite composite = (Composite) children [i];
+					Composite composite = (Composite) child;
 					Control [] grandchildren = composite.getChildren ();
-					for (int j=0; j<grandchildren.length; j++) {
-						if (grandchildren [j] instanceof Button) {
-							Button button = (Button) grandchildren [j];
+					for (Control grandchild : grandchildren) {
+						if (grandchild instanceof Button) {
+							Button button = (Button) grandchild;
 							button.addSelectionListener (selectionListener);
 						}
 					}
@@ -1314,8 +1314,7 @@ abstract class Tab {
 	
 	void setExampleWidgetPopupMenu() {
 		Control[] controls = getExampleControls();
-		for (int i = 0; i < controls.length; i++) {
-			final Control control = controls [i];
+		for (final Control control : controls) {
 			control.addListener(SWT.MenuDetect, new Listener() {
 				public void handleEvent(Event event) {
 		        	Menu menu = control.getMenu();
@@ -1344,8 +1343,8 @@ abstract class Tab {
 	 */
 	void disposeExampleWidgets () {
 		Widget [] widgets = getExampleWidgets ();
-		for (int i=0; i<widgets.length; i++) {
-			widgets [i].dispose ();
+		for (Widget widget : widgets) {
+			widget.dispose ();
 		}
 	}
 	
@@ -1451,12 +1450,12 @@ abstract class Tab {
 	Control [] getExampleControls () {
 		Widget [] widgets = getExampleWidgets ();
 		Control [] controls = new Control [0];
-		for (int i = 0; i < widgets.length; i++) {
-			if (widgets[i] instanceof Control) {
+		for (Widget widget : widgets) {
+			if (widget instanceof Control) {
 				Control[] newControls = new Control[controls.length + 1];
 				System.arraycopy(controls, 0, newControls, 0, controls.length);
 				controls = newControls;
-				controls[controls.length - 1] = (Control)widgets[i];
+				controls[controls.length - 1] = (Control)widget;
 			}
 		}
 		return controls;
@@ -1496,12 +1495,12 @@ abstract class Tab {
 	void hookExampleWidgetListeners () {
 		if (logging) {
 			Widget[] widgets = getExampleWidgets ();
-			for (int i = 0; i < widgets.length; i++) {
-				hookListeners (widgets [i]);
+			for (Widget widget : widgets) {
+				hookListeners (widget);
 			}
 			Item[] exampleItems = getExampleWidgetItems ();
-			for (int i = 0; i < exampleItems.length; i++) {
-				hookListeners (exampleItems [i]);
+			for (Item exampleItem : exampleItems) {
+				hookListeners (exampleItem);
 			}
 			String [] customNames = getCustomEventNames ();
 			for (int i = 0; i < customNames.length; i++) {
@@ -1699,8 +1698,8 @@ abstract class Tab {
 		if (colorAndFontTable == null) return; // user cannot change color/font on this tab
 		Control [] controls = getExampleControls ();
 		if (!instance.startup) {
-			for (int i = 0; i < controls.length; i++) {
-				controls[i].setBackground (backgroundColor);
+			for (Control control : controls) {
+				control.setBackground (backgroundColor);
 			}
 		}
 		// Set the background color item's image to match the background color of the example widget(s).
@@ -1718,8 +1717,8 @@ abstract class Tab {
 	 */
 	void setExampleWidgetEnabled () {
 		Control [] controls = getExampleControls ();
-		for (int i=0; i<controls.length; i++) {
-			controls [i].setEnabled (enabledButton.getSelection ());
+		for (Control control : controls) {
+			control.setEnabled (enabledButton.getSelection ());
 		}
 	}
 	
@@ -1730,8 +1729,8 @@ abstract class Tab {
 		if (colorAndFontTable == null) return; // user cannot change color/font on this tab
 		Control [] controls = getExampleControls ();
 		if (!instance.startup) {
-			for (int i = 0; i < controls.length; i++) {
-				controls[i].setFont(font);
+			for (Control control : controls) {
+				control.setFont(font);
 			}
 		}
 		/* Set the font item's image and font to match the font of the example widget(s). */
@@ -1753,8 +1752,8 @@ abstract class Tab {
 		if (colorAndFontTable == null) return; // user cannot change color/font on this tab
 		Control [] controls = getExampleControls ();
 		if (!instance.startup) {
-			for (int i = 0; i < controls.length; i++) {
-				controls[i].setForeground (foregroundColor);
+			for (Control control : controls) {
+				control.setForeground (foregroundColor);
 			}
 		}
 		/* Set the foreground color item's image to match the foreground color of the example widget(s). */
@@ -1778,13 +1777,13 @@ abstract class Tab {
 		if (smallButton.getSelection()) size = SMALL_SIZE;
 		if (largeButton.getSelection()) size = LARGE_SIZE;
 		Control [] controls = getExampleControls ();
-		for (int i=0; i<controls.length; i++) {
+		for (Control control : controls) {
 			GridData gridData = new GridData(size, size); 
 			gridData.grabExcessHorizontalSpace = fillHButton.getSelection();
 			gridData.grabExcessVerticalSpace = fillVButton.getSelection();
 			gridData.horizontalAlignment = fillHButton.getSelection() ? SWT.FILL : SWT.LEFT;
 			gridData.verticalAlignment = fillVButton.getSelection() ? SWT.FILL : SWT.TOP;
-			controls [i].setLayoutData (gridData);
+			control.setLayoutData (gridData);
 		}
 		tabFolderPage.layout (controls);
 	}
@@ -1820,8 +1819,8 @@ abstract class Tab {
 	 */
 	void setExampleWidgetVisibility () {
 		Control [] controls = getExampleControls ();
-		for (int i=0; i<controls.length; i++) {
-			controls [i].setVisible (visibleButton.getSelection ());
+		for (Control control : controls) {
+			control.setVisible (visibleButton.getSelection ());
 		}
 	}
 
@@ -1831,8 +1830,8 @@ abstract class Tab {
 	void setExampleWidgetBackgroundImage () {
 		if (backgroundImageButton != null && backgroundImageButton.isDisposed()) return;
 		Control [] controls = getExampleControls ();
-		for (int i=0; i<controls.length; i++) {
-			controls [i].setBackgroundImage (backgroundImageButton.getSelection () ? instance.images[ControlExample.ciBackground] : null);
+		for (Control control : controls) {
+			control.setBackgroundImage (backgroundImageButton.getSelection () ? instance.images[ControlExample.ciBackground] : null);
 		}
 	}
 	
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
index 4ae3925..9dddc65 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
@@ -141,8 +141,7 @@ class TabFolderTab extends Tab {
 	}
 
 	TabItem findItem(String value, TabItem[] items) {
-		for (int i = 0; i < items.length; i++) {
-			TabItem item = items[i];
+		for (TabItem item : items) {
 			if (item.getText().equals(value)) return item;
 		}
 		return null;
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
index 755617b..3edb234 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TableTab.java
@@ -439,8 +439,7 @@ class TableTab extends ScrollableTab {
 	}
 
 	TableItem findItem(String value, TableItem[] items) {
-		for (int i = 0; i < items.length; i++) {
-			TableItem item = items[i];
+		for (TableItem item : items) {
 			if (item.getText().equals(value)) return item;
 		}
 		return null;
@@ -595,8 +594,8 @@ class TableTab extends ScrollableTab {
 	void setColumnsMoveable () {
 		boolean selection = moveableColumns.getSelection();
 		TableColumn[] columns = table1.getColumns();
-		for (int i = 0; i < columns.length; i++) {
-			columns[i].setMoveable(selection);
+		for (TableColumn column : columns) {
+			column.setMoveable(selection);
 		}
 	}
 
@@ -606,8 +605,8 @@ class TableTab extends ScrollableTab {
 	void setColumnsResizable () {
 		boolean selection = resizableColumns.getSelection();
 		TableColumn[] columns = table1.getColumns();
-		for (int i = 0; i < columns.length; i++) {
-			columns[i].setResizable(selection);
+		for (TableColumn column : columns) {
+			column.setResizable(selection);
 		}
 	}
 
@@ -682,9 +681,9 @@ class TableTab extends ScrollableTab {
 			}
 		} else {
 			table1.setSortDirection (SWT.NONE);
-			for (int j = 0; j < columns.length; j++) {
-				SelectionListener listener = (SelectionListener)columns[j].getData("SortListener");	//$NON-NLS-1$
-				if (listener != null) columns[j].removeSelectionListener(listener);
+			for (TableColumn column : columns) {
+				SelectionListener listener = (SelectionListener)column.getData("SortListener");	//$NON-NLS-1$
+				if (listener != null) column.removeSelectionListener(listener);
 			}
 		}
 	}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
index d68a8f8..c4b8ca3 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java
@@ -306,10 +306,10 @@ class TreeTab extends ScrollableTab {
 		tree1 = new Tree (treeGroup, style);
 		boolean multiColumn = multipleColumns.getSelection();
 		if (multiColumn) {
-			for (int i = 0; i < columnTitles.length; i++) {
+			for (String columnTitle : columnTitles) {
 				TreeColumn treeColumn = new TreeColumn(tree1, SWT.NONE);
-				treeColumn.setText(columnTitles[i]);
-				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitles[i]}));
+				treeColumn.setText(columnTitle);
+				treeColumn.setToolTipText(ControlExample.getResourceString("Tooltip", new String [] {columnTitle}));
 			}
 			tree1.setSortColumn(tree1.getColumn(0));
 		}
@@ -493,8 +493,7 @@ class TreeTab extends ScrollableTab {
 	}
 
 	TreeItem findItem(String value, TreeItem[] items) {
-		for (int i = 0; i < items.length; i++) {
-			TreeItem item = items[i];
+		for (TreeItem item : items) {
 			if (item.getText().equals(value)) return item;
 			item = findItem(value, item.getItems());
 			if (item != null) return item;
@@ -526,12 +525,12 @@ class TreeTab extends ScrollableTab {
 	void setColumnsMoveable () {
 		boolean selection = moveableColumns.getSelection();
 		TreeColumn[] columns1 = tree1.getColumns();
-		for (int i = 0; i < columns1.length; i++) {
-			columns1[i].setMoveable(selection);
+		for (TreeColumn column : columns1) {
+			column.setMoveable(selection);
 		}
 		TreeColumn[] columns2 = tree2.getColumns();
-		for (int i = 0; i < columns2.length; i++) {
-			columns2[i].setMoveable(selection);
+		for (TreeColumn column : columns2) {
+			column.setMoveable(selection);
 		}
 	}
 
@@ -541,12 +540,12 @@ class TreeTab extends ScrollableTab {
 	void setColumnsResizable () {
 		boolean selection = resizableColumns.getSelection();
 		TreeColumn[] columns1 = tree1.getColumns();
-		for (int i = 0; i < columns1.length; i++) {
-			columns1[i].setResizable(selection);
+		for (TreeColumn column : columns1) {
+			column.setResizable(selection);
 		}
 		TreeColumn[] columns2 = tree2.getColumns();
-		for (int i = 0; i < columns2.length; i++) {
-			columns2[i].setResizable(selection);
+		for (TreeColumn column : columns2) {
+			column.setResizable(selection);
 		}
 	}
 
@@ -778,9 +777,9 @@ class TreeTab extends ScrollableTab {
 	void resetSortState (final Tree tree) {
 		tree.setSortDirection (SWT.NONE);
 		TreeColumn [] columns = tree.getColumns();
-		for (int i = 0; i < columns.length; i++) {
-			SelectionListener listener = (SelectionListener)columns[i].getData("SortListener");	//$NON-NLS-1$
-			if (listener != null) columns[i].removeSelectionListener(listener);
+		for (TreeColumn column : columns) {
+			SelectionListener listener = (SelectionListener)column.getData("SortListener");	//$NON-NLS-1$
+			if (listener != null) column.removeSelectionListener(listener);
 		}
 	}
 	
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
index 46f996b..a50c554 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
@@ -668,8 +668,8 @@ public class FileViewer {
 	 */
 	private static void treeItemRemoveAll(TreeItem treeItem) {
 		final TreeItem[] children = treeItem.getItems();
-		for (int i = 0; i < children.length; ++i) {
-			children[i].dispose();
+		for (TreeItem child : children) {
+			child.dispose();
 		}
 	}
 
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
index 7afa68a..f2a049e 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
@@ -102,8 +102,7 @@ class IconCache {
 	 */
 	public void freeResources() {
 		if (stockImages != null) {
-			for (int i = 0; i < stockImages.length; ++i) {
-				final Image image = stockImages[i];
+			for (Image image : stockImages) {
 				if (image != null) image.dispose();
 			}
 			stockImages = null;
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CustomFontTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CustomFontTab.java
index 275f24b..3c7ee92 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CustomFontTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CustomFontTab.java
@@ -95,8 +95,7 @@ public void createControlPanel(Composite parent) {
 	
 	new Label(comp, SWT.LEFT).setText(GraphicsExample.getResourceString("FontFace")); //$NON-NLS-1$
 	fontFaceCb = new Combo(comp, SWT.DROP_DOWN);
-	for (int i=0; i < fontNames.size(); i++) {
-		String name = fontNames.get(i);
+	for (String name : fontNames) {
 		fontFaceCb.add(name);
 	}
 	fontFaceCb.select(0);
@@ -112,8 +111,8 @@ public void createControlPanel(Composite parent) {
 	
 	new Label(comp, SWT.LEFT).setText(GraphicsExample.getResourceString("FontStyle")); //$NON-NLS-1$
 	fontStyleCb = new Combo(comp, SWT.DROP_DOWN);
-	for (int i=0; i < fontStyles.length; i++) {
-		fontStyleCb.add(fontStyles[i]);
+	for (String fontStyle : fontStyles) {
+		fontStyleCb.add(fontStyle);
 	}
 	fontStyleCb.select(0);
 	fontStyleCb.addListener(SWT.Selection, new Listener() {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/GraphicsExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/GraphicsExample.java
index 30b8f04..d58cded 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/GraphicsExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/GraphicsExample.java
@@ -375,24 +375,20 @@ void createTabList(Composite parent) {
 		}
 	});
 	HashSet<String> set = new HashSet<String>();
-	for (int i = 0; i < tabs.length; i++) {
-		GraphicsTab tab = tabs[i];
+	for (GraphicsTab tab : tabs) {
 		set.add(tab.getCategory());
 	}
 	String[] categories = new String[set.size()];
 	set.toArray(categories);
 	Arrays.sort(categories);
-	for (int i = 0; i < categories.length; i++) {
-		String text = categories[i];
+	for (String text : categories) {
 		TreeItem item = new TreeItem(tabList, SWT.NONE);
 		item.setText(text);
 	}
 	tabs_in_order = new ArrayList<GraphicsTab>();
 	TreeItem[] items = tabList.getItems();
-	for (int i = 0; i < items.length; i++) {
-		TreeItem item = items[i];
-		for (int j = 0; j < tabs.length; j++) {
-			GraphicsTab tab = tabs[j];
+	for (TreeItem item : items) {
+		for (GraphicsTab tab : tabs) {
 			if (item.getText().equals(tab.getCategory())) {
 				TreeItem item1 = new TreeItem(item, SWT.NONE);
 				item1.setText(tab.getText());
@@ -466,16 +462,15 @@ GraphicsTab[] createTabs() {
  */
 public void dispose() {
 	if (tabs != null) {
-		for (int i = 0; i < tabs.length; i++) {
-			GraphicsTab tab = tabs[i];
+		for (GraphicsTab tab : tabs) {
 			tab.dispose();
 		}
 	}
 	tabs = null;
 	if (resources != null) {
-		for (int i = 0; i < resources.size(); i++) {
-			if (resources.get(i) != null) {
-				resources.get(i).dispose();
+		for (Image image : resources) {
+			if (image != null) {
+				image.dispose();
 			}
 		}
 	}
@@ -488,8 +483,7 @@ public void dispose() {
 }
 
 TreeItem findItemByData(TreeItem[] items, Object data) {
-	for (int i = 0; i < items.length; i++) {
-		TreeItem item = items[i];
+	for (TreeItem item : items) {
 		if (item.getData() == data) return item;
 		item = findItemByData(item.getItems(), data);
 		if (item != null) return item;
@@ -580,8 +574,7 @@ public void setFocus() {
  */
 public void setTab(GraphicsTab tab) {
 	Control[] children = tabControlPanel.getChildren();
-	for (int i = 0; i < children.length; i++) {
-		Control control = children[i];
+	for (Control control: children) {
 		control.dispose();
 	}
 	if (this.tab != null) this.tab.dispose();
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
index a6540dc..91c739c 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
@@ -272,8 +272,8 @@ public class ImageAnalyzer {
 		group.setText(bundle.getString("Save_group"));
 		imageTypeCombo = new Combo(group, SWT.DROP_DOWN | SWT.READ_ONLY);
 		String[] types = {"JPEG", "PNG", "GIF", "ICO", "TIFF", "BMP"};
-		for (int i = 0; i < types.length; i++) {
-			imageTypeCombo.add(types[i]);
+		for (String type : types) {
+			imageTypeCombo.add(type);
 		}
 		imageTypeCombo.select(imageTypeCombo.indexOf("JPEG"));
 		imageTypeCombo.addSelectionListener(new SelectionAdapter() {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaLineStyler.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaLineStyler.java
index 094ca64..a186742 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaLineStyler.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaLineStyler.java
@@ -253,8 +253,8 @@ public class JavaScanner {
 	void initialize() {
 		fgKeys= new Hashtable<String, Integer>();
 		Integer k= new Integer(KEY);
-		for (int i= 0; i < fgKeywords.length; i++)
-			fgKeys.put(fgKeywords[i], k);
+		for (String word : fgKeywords)
+			fgKeys.put(word, k);
 	}
 
 	/**
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/layoutexample/LayoutExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/layoutexample/LayoutExample.java
index 2d697d9..71e687b 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/layoutexample/LayoutExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/layoutexample/LayoutExample.java
@@ -38,10 +38,10 @@ public class LayoutExample {
 			new FormLayoutTab (this),
 			new StackLayoutTab (this)
 		};
-		for (int i=0; i<tabs.length; i++) {
+		for (Tab tab : tabs) {
 			TabItem item = new TabItem (tabFolder, SWT.NONE);
-		    item.setText (tabs [i].getTabText ());
-		    item.setControl (tabs [i].createTabFolderPage (tabFolder));
+		    item.setText (tab.getTabText ());
+		    item.setControl (tab.createTabFolderPage (tabFolder));
 		}
 	}
 	
